planner/core/expression_rewriter.go

```diff
@@ -203,11 +203,7 @@ func (er *expressionRewriter) constructBinaryOpFunction(l expression.Expression,
 	default:
 		larg0, rarg0 := expression.GetFuncArg(l, 0), expression.GetFuncArg(r, 0)
 		var expr1, expr2, expr3 expression.Expression
-		if op == ast.LE || op == ast.GE {
-			expr1 = expression.NewFunctionInternal(er.ctx, op, types.NewFieldType(mysql.TypeTiny), larg0, rarg0)
-			expr1 = expression.NewFunctionInternal(er.ctx, ast.EQ, types.NewFieldType(mysql.TypeTiny), expr1, expression.Zero)
-			expr2 = expression.Zero
-		} else if op == ast.LT || op == ast.GT {
+		if op == ast.LT || op == ast.GT || op == ast.LE || op == ast.GE {
```

Comment:
This check can be removed?

Reply:
Is there any other operator?

Reply:
I'm not really sure about this.
But expr1 and expr2 will be used out of this check block,
if there are other operators, it may cause unexpected errors.
You may be better to confirm it.

Reply:
ok

Reply:
@XuHuaiyu addressed